src/main/java/lotto/domain/WinningInfo.java

```diff
@@ -11,16 +11,24 @@ private WinningInfo(Map<Rank, Integer> winningInfo) {
         this.winningInfo = winningInfo;
     }
 
-    public static WinningInfo of(PickedLottoNumbers pickedLottoNumbers, LottoNumbers winningNumbers) {
+    public static WinningInfo of(Lottos lottos, LottoNumber bonusNumber, Lotto winningNumbers) {
         HashMap<Rank, Integer> winningInfo = new HashMap<>();
-        for (LottoNumbers pickedNumber : pickedLottoNumbers) {
-            Rank rank = Rank.findRank(compareLottoNumbers(pickedNumber, winningNumbers));
+
+        for (Lotto pickedNumber : lottos) {
+            int matchCount = compareLottoNumbers(pickedNumber, winningNumbers);
+            boolean matchBonus = containBonusNumber(winningNumbers, bonusNumber);
+
+            Rank rank = Rank.findRank(matchCount, matchBonus);
```

Comment:
이 부분 책임 자체를 아래처럼 Lotto로 넘길 순 없을까요?

```suggestion
            Rank rank = pickedNumber.getRank(winningLotto);
```

Reply:
당첨 정보에서 몇등인지 뽑는게 맞다고 생각했었는데, 피드백 듣고 다시 생각해보니 등수는 로또를 통해 알 수 있으니 로또에서 등수를 알려주는게 더 합리적이라는 생각이 드네요. 감사합니다!